Code/ElastixTransformixWrappers/include/sitkTransformixImageFilter.h

```diff
@@ -122,7 +123,7 @@ class SITKElastix_EXPORT TransformixImageFilter
 private:
 
   class TransformixImageFilterImpl;
-  TransformixImageFilterImpl* m_Pimple;
+  const std::unique_ptr<TransformixImageFilterImpl> m_Pimple;
```

Comment:
We have previously had problems using forward declared/undefined classes as arguments to these types of smart pointers.

Reply:
Thanks for your comment, @blowekamp Do you still remember what kind of problems? Because `std::unique_ptr` is really _meant_ to support forward declared Pimpl classes, as shown by the Pimpl example in the C++ Core Guidelines, at https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Ri-pimpl

In order for this to work, the destructor of the exported class (`TransformixImageFilter` in our case) must be implemented in the CXX file (or CPP file), not in the header file. Or more precisely, the Impl class must be fully defined (rather than just forward-declared) at the point of the destructor implementation. As `~TransformixImageFilter()` is implemented in de CXX file that does `#include "sitkTransformixImageFilterImpl.h"`, it should be just fine.  OK?